src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -64,25 +97,45 @@ public class CON_DoNotContact_TDTM extends TDTM_Runnable {
                 //deceased and do not contact: updates related fields for both checking and unchecking the box.
                 //prefers deceased as the deceased already marks the Do Not Contact fields, in case the user changes both values.
                 if (con.Deceased__c != oldCon.Deceased__c)
-                    markContactDeceased(con);
+                    markContactDeceased(con, acc);
                 else if (con.Do_Not_Contact__c != oldCon.Do_Not_Contact__c)
                     markContactDoNotContact(con);
             }
         }
         
-        return new DmlWrapper();
+        return dmlWrapper;
 	}
     
     /*******************************************************************************************************
-    * @description Toggles additional Contact fields as necessary when a contact is flagged as deceased. 
+    * @description Sets Do Not Contact and related fields when a contact is flagged as deceased. Removes from
+    * Household name if there are other non-deceased members in the household.
     * @param con The Contact record.
-    * @return void.  
     ********************************************************************************************************/
-    public void markContactDeceased(Contact con) {
+    public void markContactDeceased(Contact con, Account acc) {
 
         con.Do_Not_Contact__c = con.Deceased__c;
         markContactDoNotContact(con);
 
+        if (acc!=null && con.Deceased__c) {
+            list<Contact> contacts = (list<Contact>)acc.Contacts;
+            //if there are non-deceased members of the household
+            //Exclude from household name and update household's primary contact
+            if (!contacts.isEmpty()) {
+                excludeFromHHNaming(con);
+                if (con.Id != null && acc.npe01__One2OneContact__c == con.id) {
+                    acc.npe01__One2OneContact__c = contacts[0].id;
```

Comment:
are you sure that acc.Contacts doesn't hold the current contact who is being marked deceased.  I saw that you query for all the household contacts who aren't deceased, but since you do this in the BeforeInsert and BeforeUpdate, I think that contact might be returned in your query?